packages/jest-worker/src/Queue.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {QueueChildMessage} from './types';
+
+export default class Queue {
+  private data: Set<QueueChildMessage | null>;
+  private iterator: IterableIterator<QueueChildMessage | null>;
+
+  constructor() {
+    this.data = new Set();
+    this.iterator = this.data.values();
```

Comment:
@cpojer I implemented new queue base Set, it faster linkedlist we using now and clearer. Bẻoe, I tries to implemented with native Array but it is slower.